From 65d722507cd65bbf42f9c00d92c7016bc92968bf Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 6 May 2022 09:28:36 +0800
Subject: [PATCH] sw64: uapi: generate some uapi headers from generic ones
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 65d722507cd65bbf42f9c00d92c7016bc92968bf
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

These uapi headers are proved to be the same with generic version.

Herein we remove the arch-specific statfs.h and types.h to end
support for old system.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/Kbuild        |  1 +
 arch/sw_64/include/uapi/asm/ipcbuf.h      |  7 ----
 arch/sw_64/include/uapi/asm/kvm_para.h    |  7 ----
 arch/sw_64/include/uapi/asm/msgbuf.h      | 28 ----------------
 arch/sw_64/include/uapi/asm/poll.h        |  7 ----
 arch/sw_64/include/uapi/asm/posix_types.h | 18 -----------
 arch/sw_64/include/uapi/asm/sembuf.h      | 23 -------------
 arch/sw_64/include/uapi/asm/shmbuf.h      | 39 -----------------------
 arch/sw_64/include/uapi/asm/statfs.h      |  9 ------
 arch/sw_64/include/uapi/asm/types.h       | 28 ----------------
 10 files changed, 1 insertion(+), 166 deletions(-)
 delete mode 100644 arch/sw_64/include/uapi/asm/ipcbuf.h
 delete mode 100644 arch/sw_64/include/uapi/asm/kvm_para.h
 delete mode 100644 arch/sw_64/include/uapi/asm/msgbuf.h
 delete mode 100644 arch/sw_64/include/uapi/asm/poll.h
 delete mode 100644 arch/sw_64/include/uapi/asm/posix_types.h
 delete mode 100644 arch/sw_64/include/uapi/asm/sembuf.h
 delete mode 100644 arch/sw_64/include/uapi/asm/shmbuf.h
 delete mode 100644 arch/sw_64/include/uapi/asm/statfs.h
 delete mode 100644 arch/sw_64/include/uapi/asm/types.h

diff --git a/arch/sw_64/include/uapi/asm/Kbuild b/arch/sw_64/include/uapi/asm/Kbuild
index a01bfb9600ec..15700040f138 100644
--- a/arch/sw_64/include/uapi/asm/Kbuild
+++ b/arch/sw_64/include/uapi/asm/Kbuild
@@ -1,4 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
 # UAPI Header export list
 
+generic-y += kvm_para.h
 generated-y += unistd_64.h
diff --git a/arch/sw_64/include/uapi/asm/ipcbuf.h b/arch/sw_64/include/uapi/asm/ipcbuf.h
deleted file mode 100644
index 553cdb37052d..000000000000
--- a/arch/sw_64/include/uapi/asm/ipcbuf.h
+++ /dev/null
@@ -1,7 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_IPCBUF_H
-#define _UAPI_ASM_SW64_IPCBUF_H
-
-#include <asm-generic/ipcbuf.h>
-
-#endif
diff --git a/arch/sw_64/include/uapi/asm/kvm_para.h b/arch/sw_64/include/uapi/asm/kvm_para.h
deleted file mode 100644
index 3c0f9fa712ab..000000000000
--- a/arch/sw_64/include/uapi/asm/kvm_para.h
+++ /dev/null
@@ -1,7 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_KVM_PARA_H
-#define _UAPI_ASM_SW64_KVM_PARA_H
-
-#include <asm-generic/kvm_para.h>
-
-#endif
diff --git a/arch/sw_64/include/uapi/asm/msgbuf.h b/arch/sw_64/include/uapi/asm/msgbuf.h
deleted file mode 100644
index b938df3664a0..000000000000
--- a/arch/sw_64/include/uapi/asm/msgbuf.h
+++ /dev/null
@@ -1,28 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_MSGBUF_H
-#define _UAPI_ASM_SW64_MSGBUF_H
-
-/*
- * The msqid64_ds structure for sw64 architecture.
- * Note extra padding because this structure is passed back and forth
- * between kernel and user space.
- *
- * Pad space is left for:
- * - 2 miscellaneous 64-bit values
- */
-
-struct msqid64_ds {
-	struct ipc64_perm msg_perm;
-	long		msg_stime;	/* last msgsnd time */
-	long		msg_rtime;	/* last msgrcv time */
-	long		msg_ctime;	/* last change time */
-	unsigned long	msg_cbytes;	/* current number of bytes on queue */
-	unsigned long	msg_qnum;	/* number of messages in queue */
-	unsigned long	msg_qbytes;	/* max number of bytes on queue */
-	__kernel_pid_t	msg_lspid;	/* pid of last msgsnd */
-	__kernel_pid_t	msg_lrpid;	/* last receive pid */
-	unsigned long	__unused1;
-	unsigned long	__unused2;
-};
-
-#endif /* _UAPI_ASM_SW64_MSGBUF_H */
diff --git a/arch/sw_64/include/uapi/asm/poll.h b/arch/sw_64/include/uapi/asm/poll.h
deleted file mode 100644
index 114d0344e377..000000000000
--- a/arch/sw_64/include/uapi/asm/poll.h
+++ /dev/null
@@ -1,7 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_POLL_H
-#define _UAPI_ASM_SW64_POLL_H
-
-#include <asm-generic/poll.h>
-
-#endif
diff --git a/arch/sw_64/include/uapi/asm/posix_types.h b/arch/sw_64/include/uapi/asm/posix_types.h
deleted file mode 100644
index 182741aaa06e..000000000000
--- a/arch/sw_64/include/uapi/asm/posix_types.h
+++ /dev/null
@@ -1,18 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_POSIX_TYPES_H
-#define _UAPI_ASM_SW64_POSIX_TYPES_H
-
-/*
- * This file is generally used by user-level software, so you need to
- * be a little careful about namespace pollution etc.  Also, we cannot
- * assume GCC is being used.
- */
-
-typedef unsigned long	__kernel_ino_t;
-#define __kernel_ino_t	__kernel_ino_t
-
-typedef unsigned long	__kernel_sigset_t;	/* at least 32 bits */
-
-#include <asm-generic/posix_types.h>
-
-#endif /* _UAPI_ASM_SW64_POSIX_TYPES_H */
diff --git a/arch/sw_64/include/uapi/asm/sembuf.h b/arch/sw_64/include/uapi/asm/sembuf.h
deleted file mode 100644
index 08b0876e739c..000000000000
--- a/arch/sw_64/include/uapi/asm/sembuf.h
+++ /dev/null
@@ -1,23 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_SEMBUF_H
-#define _UAPI_ASM_SW64_SEMBUF_H
-
-/*
- * The semid64_ds structure for sw64 architecture.
- * Note extra padding because this structure is passed back and forth
- * between kernel and user space.
- *
- * Pad space is left for:
- * - 2 miscellaneous 64-bit values
- */
-
-struct semid64_ds {
-	struct ipc64_perm sem_perm;		/* permissions .. see ipc.h */
-	long		sem_otime;		/* last semop time */
-	long		sem_ctime;		/* last change time */
-	unsigned long	sem_nsems;		/* no. of semaphores in array */
-	unsigned long	__unused1;
-	unsigned long	__unused2;
-};
-
-#endif /* _UAPI_ASM_SW64_SEMBUF_H */
diff --git a/arch/sw_64/include/uapi/asm/shmbuf.h b/arch/sw_64/include/uapi/asm/shmbuf.h
deleted file mode 100644
index 4572337bee02..000000000000
--- a/arch/sw_64/include/uapi/asm/shmbuf.h
+++ /dev/null
@@ -1,39 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_SHMBUF_H
-#define _UAPI_ASM_SW64_SHMBUF_H
-
-/*
- * The shmid64_ds structure for sw64 architecture.
- * Note extra padding because this structure is passed back and forth
- * between kernel and user space.
- *
- * Pad space is left for:
- * - 2 miscellaneous 64-bit values
- */
-
-struct shmid64_ds {
-	struct ipc64_perm	shm_perm;	/* operation perms */
-	size_t			shm_segsz;	/* size of segment (bytes) */
-	long			shm_atime;	/* last attach time */
-	long			shm_dtime;	/* last detach time */
-	long			shm_ctime;	/* last change time */
-	__kernel_pid_t		shm_cpid;	/* pid of creator */
-	__kernel_pid_t		shm_lpid;	/* pid of last operator */
-	unsigned long		shm_nattch;	/* no. of current attaches */
-	unsigned long		__unused1;
-	unsigned long		__unused2;
-};
-
-struct shminfo64 {
-	unsigned long	shmmax;
-	unsigned long	shmmin;
-	unsigned long	shmmni;
-	unsigned long	shmseg;
-	unsigned long	shmall;
-	unsigned long	__unused1;
-	unsigned long	__unused2;
-	unsigned long	__unused3;
-	unsigned long	__unused4;
-};
-
-#endif /* _UAPI_ASM_SW64_SHMBUF_H */
diff --git a/arch/sw_64/include/uapi/asm/statfs.h b/arch/sw_64/include/uapi/asm/statfs.h
deleted file mode 100644
index b92d719238d1..000000000000
--- a/arch/sw_64/include/uapi/asm/statfs.h
+++ /dev/null
@@ -1,9 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_STATFS_H
-#define _UAPI_ASM_SW64_STATFS_H
-
-#include <linux/types.h>
-
-#include <asm-generic/statfs.h>
-
-#endif
diff --git a/arch/sw_64/include/uapi/asm/types.h b/arch/sw_64/include/uapi/asm/types.h
deleted file mode 100644
index 750b5181c3de..000000000000
--- a/arch/sw_64/include/uapi/asm/types.h
+++ /dev/null
@@ -1,28 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_TYPES_H
-#define _UAPI_ASM_SW64_TYPES_H
-
-/*
- * This file is never included by application software unless
- * explicitly requested (e.g., via linux/types.h) in which case the
- * application is Linux specific so (user-) name space pollution is
- * not a major issue.  However, for interoperability, libraries still
- * need to be careful to avoid a name clashes.
- */
-
-/*
- * This is here because we used to use l64 for sw64 and we don't want
- * to impact user mode with our change to ll64 in the kernel.
- *
- * However, some user programs are fine with this.  They can
- * flag __SANE_USERSPACE_TYPES__ to get int-ll64.h here.
- */
-#ifndef __KERNEL__
-#ifndef __SANE_USERSPACE_TYPES__
-#include <asm-generic/int-l64.h>
-#else
-#include <asm-generic/int-ll64.h>
-#endif /* __SANE_USERSPACE_TYPES__ */
-#endif /* __KERNEL__ */
-
-#endif /* _UAPI_ASM_SW64_TYPES_H */
-- 
2.33.0

